File: src/app/models/SoftwareUpdateCampaign.js


// Define collection and schema for SoftwareUpdateCampaign
export interface SoftwareUpdateCampaign {
    campaignCode:
	type : string
    scheduledStart:
	type : Date
    scheduledEnd:
	type : Date
    FirmwareRelease:
	type : Schema.Types.ObjectId
    DeviceGroup:
	type : Schema.Types.ObjectId
    Executions:
 	type : [{ type: Schema.Types.ObjectId, ref: 'SoftwareUpdateExecution' }]
    Status:
 	type : String
#
    collection: 'softwareUpdateCampaigns'
}

File: iotonaspdotnet/Domain/SoftwareUpdateCampaign.cs
using iotonaspdotnet.Domain.Contracts;

namespace iotonaspdotnet.Domain;

public class SoftwareUpdateCampaign
{
    public Guid Id { get; set; } = Guid.NewGuid();

 public virtual long SoftwareupdatecampaignId { get; set; }
 public virtual string CampaignCode { get; set; }
 public virtual DateTime ScheduledStart { get; set; }
 public virtual DateTime ScheduledEnd { get; set; }
public virtual FirmwareRelease FirmwareRelease { get; set; }
public virtual DeviceGroup DeviceGroup { get; set; }
public virtual SoftwareUpdateExecution Executions { get; set; }
 public virtual UpdateCampaignStatus Status { get; set; }

    public static SoftwareUpdateCampaign FromRequest(SoftwareUpdateCampaignRequest request) {
        return new SoftwareUpdateCampaign {
            Id = model.Id,
            CampaignCode = request.CampaignCode,
            ScheduledStart = request.ScheduledStart,
            ScheduledEnd = request.ScheduledEnd,
            Status = request.Status,
        };
}
